src/content_cao.cpp

```diff
@@ -1814,6 +1832,30 @@ bool GenericCAO::directReportPunch(v3f dir, const ItemStack *punchitem,
 	return false;
 }
 
+bool GenericCAO::isNotItemStack() const
+{
+	// We know what is needed for something *not* to be
+	// an itemstack. However, we don't know for sure
+	// whether anything *is* an itemstack.
+	// (hence the check for 'not an itemstack')
+	static const v3f zero(0,0,0);
+	return
+		false
+		// Item stack may be physical or non-physical...
+		//|| m_prop.physical
+		|| m_prop.collideWithObjects
+		|| !m_is_visible
+		|| m_prop.makes_footstep_sound
+		|| m_prop.automatic_rotate == 0
+		|| m_prop.stepheight != 0
+		|| m_prop.automatic_face_movement_dir
+		// Item stack may be falling down first
+		//|| m_velocity != zero
+		// Acceleration may be zero or non-zero
+		//|| m_acceleration != zero
+		|| m_prop.visual != "wielditem";
```

Comment:
I invstigated a bit, and my impression so far is that it can't be used to distinguish whether some active object represents a dropped itemstack. It seems that the best I can do from there is find the (or _a_) defined item that uses the same texture as the active object. If such an item exists, the active object can still be of another type. A falling node for instance.

> There also seems to be an ItemCAO, but i can't find where it is used.

IIUC, ItemCAO is deprecated. All CAOs are now lua-based.